patches/minecraft/net/minecraft/world/level/block/FlowerBlock.java.patch

```diff
@@ -0,0 +1,41 @@
+--- a/net/minecraft/world/level/block/FlowerBlock.java
++++ b/net/minecraft/world/level/block/FlowerBlock.java
+@@ -15,6 +_,17 @@
+    private final MobEffect f_53508_;
+    private final int f_53509_;
+ 
++   private final java.util.function.Supplier<MobEffect> suspiciousStewEffectSupplier;
++
++   public FlowerBlock(java.util.function.Supplier<MobEffect> suspiciousStewEffectSupplier, int p_53513_, BlockBehaviour.Properties p_53514_) {
++      super(p_53514_);
++      this.f_53508_ = null;
++      this.suspiciousStewEffectSupplier = suspiciousStewEffectSupplier;
++      this.f_53509_ = p_53513_;
++   }
++
++   /** @deprecated FORGE: Use supplier version instead */
++   @Deprecated
+    public FlowerBlock(MobEffect p_53512_, int p_53513_, BlockBehaviour.Properties p_53514_) {
+       super(p_53514_);
+       this.f_53508_ = p_53512_;
+@@ -23,6 +_,7 @@
+       } else {
+          this.f_53509_ = p_53513_ * 20;
+       }
++      this.suspiciousStewEffectSupplier = net.minecraftforge.registries.RegistryObject.create(p_53512_.getRegistryName(), net.minecraftforge.registries.ForgeRegistries.MOB_EFFECTS);
+ 
+    }
+ 
+@@ -36,10 +_,12 @@
+    }
+ 
+    public MobEffect m_53521_() {
++      if (true) return this.suspiciousStewEffectSupplier.get();
+       return this.f_53508_;
+    }
+ 
+    public int m_53522_() {
++      if (this.f_53508_ == null && !this.suspiciousStewEffectSupplier.get().m_8093_()) return this.f_53509_ * 20;
```

Comment:
Why the `== null` condition granted that it will always be true in normal circumstances?